### PR TITLE
Bug fix - Login Code clean up

### PR DESCRIPTION
**Pull request by ksookdhew**
## Description
Clean up Code for Login

## Checklist 
Please review and check the following before submitting this pull request: 
- [x] The code follows the project's coding standards. 
- [x] 100% Unit test coverage for View Models 
- [x] All warnings Resolved.

### CODE REVIEW

**Review by MarcusNkadimeng — APPROVED** [review]
Looks good

File: F1/F1/Login/LoginViewController.swift [hunk 1]
```diff
@@ -9,32 +9,27 @@ import UIKit
 
 class LoginViewController: UIViewController {
 
+    // MARK: IBOutlets
     @IBOutlet weak var username: UITextField!
     @IBOutlet weak var password: UITextField!
-    @IBOutlet weak var errorMsg: UILabel!
+    @IBOutlet weak var errorMessage: UILabel!
```
Comment (kieran-woodrow) [inline F1/F1/Login/LoginViewController.swift:15]: private for all these please

File: F1/F1/Login/LoginViewController.swift [hunk 2]
```diff
@@ -9,32 +9,27 @@ import UIKit
...
+
+    // MARK: Variables
     private lazy var viewModel = LoginViewModel()
```
Comment (kieran-woodrow) [inline F1/F1/Login/LoginViewController.swift:18]: viewModel doesn't take in any delegates or repos so can remove lazy keyword

File: F1/F1/Login/LoginViewController.swift [hunk 3]
```diff
@@ -9,32 +9,27 @@ import UIKit
...
     }
 
+    // MARK: IBAction
     @IBAction func loginAction(_ sender: Any) {
-        if viewModel.validDetails(givenUsername: username.text ?? "", givenPassword: password.text ?? "") {
-            errorMsg.isHidden = true
-            performSegue(withIdentifier: "showTabSegue", sender: self)
+        if viewModel.validDetails(givenUsername: username.text, givenPassword: password.text) {
+            errorMessage.isHidden = true
+            performSegue(withIdentifier: Identifiers.showTabSegue, sender: self)
```
Comment (kieran-woodrow) [inline F1/F1/Login/LoginViewController.swift:30]: If you want to clean it up a bit more, this perform segue can be done in the viewModel using a delegate

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

**Review by AShozi — DISMISSED** [review]

File: F1/F1/Login/LoginViewController.swift (hunk 3 above)
Comment (ksookdhew) [reply F1/F1/Login/LoginViewController.swift]: Added a delegate so no need to remove now 😄

File: F1/F1/Login/LoginViewController.swift (hunk 2 above)
Comment (ksookdhew) [reply F1/F1/Login/LoginViewController.swift]: Added a delegate, so no need to remove anymore 😄

File: F1/F1/Login/LoginViewController.swift (hunk 3 above)
Comment (ksookdhew) [reply F1/F1/Login/LoginViewController.swift]: Can you just check I did that right? 😄

**Review by kieran-woodrow — APPROVED** [review]

**Review by giacatano — APPROVED** [review]

**Pull request merged**